VirtoCommerce.CatalogModule.Web/Scripts/directives/property2.js

```diff
@@ -116,28 +115,57 @@ angular.module('virtoCommerce.catalogModule')
                 }
             };
 
-            function loadDictionaryValues() {
-                scope.getPropValues()(scope.currentEntity.id).then(function (result) {
-                    scope.context.allDictionaryValues = [];
-                    scope.context.currentPropValues = [];
+            scope.loadDictionaryValues = function ($select) {
+                $select.page = 0;
+                scope.context.allDictionaryValues = [];
 
-                    angular.forEach(result.results, function (dictItem) {
-                        var dictValue = { alias: dictItem.alias, valueId: dictItem.id, value: dictItem.alias };
-                        //Need to select already selected values. Dictionary values have same type as standard values.
-                        dictValue.selected = angular.isDefined(_.find(scope.currentEntity.values, function (x) { return x.valueId == dictItem.id }));
-                        scope.context.allDictionaryValues.push(dictValue);
-                        if (dictValue.selected) {
-                            //add selected value
-                            scope.context.currentPropValues.push(dictValue);
-                        }
-                    });
+                return scope.loadNextDictionaryValues($select);
+            };
+
+            scope.loadNextDictionaryValues = function($select) {
+                var countToSkip = $select.page * scope.pageSize;
+                var countToTake = scope.pageSize;
 
-                    //initLanguagesValuesMap();
+                return scope.getPropValues()(scope.currentEntity.id, '', countToSkip, countToTake).then(function (result) {
```

Comment:
Search keyword doesn't take into accounts, it will be difficult to find appropriate dictionary value in the huge amount of data

Reply:
Sorry, I forgot about this. It is fixed now.

I also reduced `refresh-timeout` to 200 ms (default value was 1000 ms), so now search keyword will be applied quicker.